runner/src/run.rs

```diff
@@ -54,6 +57,22 @@ pub fn run(args: &RunArgs) {
             FIRMWARE_ADDR
         ));
 
+    if let Some(payload) = &cfg.target.payload {
```

Comment:
Maybe add a small comment above to explain what this block of code is doing (load the payload if any).